File: app/src/main/java/com/kludgenics/cgmlogger/model/nightscout/NightscoutApiEndpoint.java
package com.kludgenics.cgmlogger.model.nightscout;

import java.util.List;
import java.util.Map;

import retrofit.http.GET;
import retrofit.http.Query;

/**
 * Created by matthiasgranberry on 6/3/15.
 */
public interface NightscoutApiEndpoint {
    @GET("/api/v1/entries.json")
    List<NightscoutApiEntry> getEntries(@Query("count") int count);

    @GET("/api/v1/entries/sgv.json")
    List<SgvEntry> getSgvEntries(@Query("count") int count, @Query("find[dateString][$gte]") String start);

    @GET("/api/v1/treatments")
    List<Map<String, String>> getTreatments();

}

File: app/src/main/java/com/kludgenics/cgmlogger/model/nightscout/NightscoutApiEndpoint.kt
package com.kludgenics.cgmlogger.model.nightscout

import retrofit.Call
import retrofit.http.GET
import retrofit.http.Query
import java.util.*

/**
 * Created by matthiasgranberry on 6/3/15.
 */
interface NightscoutApiEndpoint {
    @GET("/api/v1/entries/sgv.json")
    fun getSgvEntries(@Query("count") count: Int, @Query("find[dateString][\$gte]") start: String): Call<ArrayList<SgvEntry>>

    @GET("/api/v1/treatments")
    fun getTreatmentsSince(@Query("find[created_at][\$gte]") start: String): Call<ArrayList<LinkedHashMap<String, String>>>
}